AmplifyPlugins/Core/AWSPluginsCore/Model/Support/QueryPredicate+GraphQL.swift

```diff
@@ -46,12 +66,36 @@ public struct GraphQLFilterConverter {
 
 /// Extension to translate a `QueryPredicate` into a GraphQL query variables object
 extension QueryPredicate {
+    public func getGraphQLFilter(_ modelSchema: ModelSchema) -> GraphQLFilter {
```

Comment:
This function seems to be exactly the same as `graphQLFilterInternal` -- it seems that `graphQLFilterInternal` is now required since it is part of the `GraphQLFilterConvertible` protocol...

We do we need to introduce getGraphQLFilter?

Reply:
```swift
protocol GraphQLFilterConvertible {
    func graphQLFilterInternal(_ modelSchema: ModelSchema?) -> GraphQLFilter
}
```
the `graphQLFilterInternal(_ modelSchema: ModelSchema? = nil)` takes `ModelSchema` as optional, but we want the public facing one `getGraphQLFilter` to take `ModelSchema` as required, in this way, the `getGraphQLFilter(_ modelSchema: ModelSchema)` can call `graphQLFilterInternal(_ modelSchema: ModelSchema? = nil)`.

And the reason to have a method that takes optional `ModelSchema` is deprecated `graphQLFilter` in `extension QueryPredicate` used to be a public computed property, better not to remove it immediately but to deprecate it. And it should call `graphQLFilterInternal()` with passing nil `ModelSchema`

Reply:
But I agree this is duplicate
I have changed it to
```swift
public func getGraphQLFilter(_ modelSchema: ModelSchema) -> GraphQLFilter {
    return graphQLFilterInternal(modelSchema)
}
```

Reply:
If the implementation of `graphQLFilter` is broken, does it make sense to keep it around?  Seems to me it would be better to remove it immediately.

Reply:
Discussed with team offline, we will deprecate it first and at some point completely remove it